dds/DCPS/LinuxNetworkConfigPublisher.cpp

```diff
@@ -0,0 +1,243 @@
+/*
+ *
+ *
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "DCPS/DdsDcps_pch.h" //Only the _pch include should start with DCPS/
+
+#include "ace/config.h"
+
+#ifdef ACE_LINUX
+
+#include "LinuxNetworkConfigPublisher.h"
+
+#include <ace/Netlink_Addr.h>
+
+#include <linux/rtnetlink.h>
+#include <net/if.h>
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+LinuxNetworkConfigPublisher::LinuxNetworkConfigPublisher(ReactorInterceptor_rch interceptor)
+{
+  reactor(interceptor->reactor());
+}
+
+bool LinuxNetworkConfigPublisher::open()
+{
+  // Listener to changes in links and IPV4 addresses.
+  ACE_Netlink_Addr addr;
+  addr.set(getpid(), RTMGRP_NOTIFY | RTMGRP_LINK | RTMGRP_IPV4_IFADDR);
+  if (socket_.open(addr, AF_NETLINK, NETLINK_ROUTE) != 0) {
+    ACE_ERROR((LM_ERROR, "LinuxNetworkConfigPublisher::open: could not open socket: %m\n"));
```

Comment:
Prefer to have some form of "error" in the message itself so that if logging is not set to verbose the user still realizes that it's an error.